tools/compiletest/src/common.rs

```diff
@@ -19,6 +19,7 @@ pub enum Mode {
     KaniFixme,
     CargoKani,
     CargoKaniTest, // `cargo kani --tests`. This is temporary and should be removed when s2n-quic moves --tests to `Cargo.toml`.
+    CoverageBased,
```

Comment:
I still don't understand why we are adding this here... But since we still need to improve how we test coverage, we  can handle this later.

Reply:
Having a dedicated mode for coverage seems OK to me. Or am I missing something here?

Last night I was thinking of some improvements that we could make for this mode (e.g., annotate lines with a comment that includes the status), but it doesn't make sense to invest a lot on it since we'll change how we represent results soon.